File: research/presentation/workspace/make_gradient.py
from PIL import Image, ImageDraw
import os

out_dir = r"C:/Users/Batu/research-template/research/presentation/workspace"

# Slide dimensions in pixels (720x405, matching pt dimensions)
W, H = 720, 405

# Create RGBA image (transparent background)
img = Image.new("RGBA", (W, H), (0, 0, 0, 0))
draw = ImageDraw.Draw(img)

# Draw the diagonal stripe as a filled polygon, then apply gradient alpha
# Stripe: from top-left area sweeping to bottom-right, then fade alpha row by row
# Polygon shape: top-left triangle/parallelogram
# Points: (0,0), (420,0), (180,405), (0,405)
polygon = [(0, 0), (420, 0), (180, 405), (0, 405)]

# Create a separate layer for the stripe
stripe = Image.new("RGBA", (W, H), (0, 0, 0, 0))
stripe_draw = ImageDraw.Draw(stripe)

# Draw the polygon in electric blue
stripe_draw.polygon(polygon, fill=(0, 194, 255, 255))

# Apply gradient alpha: fade from left to right within the polygon
# We'll do this by creating a gradient mask
import numpy as np

# Convert stripe to numpy
arr = np.array(stripe, dtype=np.float32)

# For each pixel, compute distance from left edge relative to the polygon width
# Simple approach: gradient based on x coordinate
# At x=0: alpha ~0.18 * 255 = 46
# At x=420: alpha ~0.0
# Beyond x=420: already transparent (polygon ends)

x_coords = np.arange(W, dtype=np.float32)
# alpha_factor: 0.18 at x=0, 0.0 at x=420, 0 beyond
alpha_factor = np.clip(1.0 - x_coords / 420.0, 0, 1) * 0.18
# Shape: (W,)
# Apply to image: arr[:, :, 3] is alpha channel
# alpha_factor broadcast over height
alpha_2d = np.tile(alpha_factor, (H, 1))  # (H, W)

# Multiply existing alpha by our factor
arr[:, :, 3] = arr[:, :, 3] * alpha_2d

# Convert back
result = Image.fromarray(arr.astype(np.uint8), "RGBA")

out_path = os.path.join(out_dir, "gradient_stripe.png")
result.save(out_path)
print(f"Saved: {out_path}")

File: research/presentation/workspace/make_gradient.js
const sharp = require('sharp');
const path = require('path');

const outDir = 'C:/Users/Batu/research-template/research/presentation/workspace';

async function main() {
  // Diagonal gradient stripe: electric blue #00C2FF fading to transparent
  // SVG: 720x405 (matches slide pt dimensions, treated as px for raster)
  // Diagonal stripe from top-left corner area going bottom-right
  const svg = `<svg xmlns="http://www.w3.org/2000/svg" width="720" height="405">
    <defs>
      <linearGradient id="diag" x1="0%" y1="0%" x2="60%" y2="100%">
        <stop offset="0%" style="stop-color:#00C2FF;stop-opacity:0.18"/>
        <stop offset="40%" style="stop-color:#00C2FF;stop-opacity:0.07"/>
        <stop offset="100%" style="stop-color:#00C2FF;stop-opacity:0.0"/>
      </linearGradient>
    </defs>
    <polygon points="0,0 420,0 180,405 0,405" fill="url(#diag)"/>
  </svg>`;

  await sharp(Buffer.from(svg))
    .png()
    .toFile(path.join(outDir, 'gradient_stripe.png'));

  console.log('gradient_stripe.png created');
}

main().catch(e => { console.error(e); process.exit(1); });
